source/fundamentals/connection/socks.txt

```diff
@@ -0,0 +1,139 @@
+     - Description
+
+   * - **proxyHost**
+     - string
+     - ``null``
+     - Specifies the IPv4/IPv6 address or domain name on which a SOCKS5
+       proxy is listening.
+
+   * - **proxyPort**
+     - non-negative integer
+     - ``null``
+     - Specifies the TCP port number of the SOCKS5 proxy server specified in the
+       ``proxyHost`` option. If you have set the ``proxyHost`` option,
+       this value of this option defaults to ``1080``.
+
+   * - **proxyUsername**
+     - string
+     - ``null``
+     - Specifies the username for username and password authentication to the SOCKS5
+       proxy server specified in the ``proxyHost`` option. If you set
+       this option to a zero-length string, the driver ignores it.
+
+   * - **proxyPassword**
+     - string
+     - ``null``
+     - Specifies the password for username and password authentication to the SOCKS5
+       proxy server specified in the ``proxyHost`` option. If you set
+       this option to a zero-length string, the driver ignores it.
+
+.. important::
+   
+   The driver will throw an error if you set the ``proxyPort``,
+   ``proxyUsername``, or ``proxyPassword`` options without setting the
+   ``proxyHost`` option.
+
+Example
+-------
+
+This example shows how to instantiate a ``MongoClient`` that uses SOCKS5
+proxy support. The example connects to MongoDB and executes a find
+query:
+
+.. code-block:: js
+   
+   import { MongoClient } from "mongodb";
+   
+   // Replace the uri string with your connection string.
+   const uri = "<connection uri>";
+   
+   // Replace the placeholders with your SOCKS5 proxy server information.
+   const socksOptions = {
+     proxyHost: "<host>",
+     proxyPort: 1010,
+     proxyUsername: "<username>",
+     proxyPassword: "<password>",
+   };
+   
+   // Create a client with the socksOptions object
+   const client = new MongoClient(uri, socksOptions);
+   
+   async function run() {
+     try {
+       const db = client.db("myDB");
+       const myColl = db.collection("myColl");
+       const doc = await myColl.findOne({});
+       console.log(doc);
+     } finally {
+       await client.close();
+     }
+   }
+   run().catch(console.dir);
+
+Additional Information
+----------------------
+
+For more information about SOCKS5 proxy support, see the
+`MongoDB SOCKS5 specification <https://github.com/mongodb/specifications/blob/master/source/socks5-support/socks5.rst>`__.
+
+API Documentation
+~~~~~~~~~~~~~~~~~
+
+- `MongoClientOptions <{+api+}/interfaces/MongoClientOptions.html>`__
+- `MongoClient <{+api+}/classes/MongoClient.html>`__
+- `ProxyOptions <{+api+}/interfaces/ProxyOptions.html>`__
+- `proxyHost client option <{+api+}/interfaces/ProxyOptions.html#proxyHost>`__
+- `proxyPassword client option <{+api+}/interfaces/ProxyOptions.html#proxyPassword>`__
+- `proxyPort client option <{+api+}/interfaces/ProxyOptions.html#proxyPort>`__
+- `proxyUsername client option <{+api+}/interfaces/ProxyOptions.html#proxyUsername>`__
```

Comment:
Suggestion:
I think these can be removed since they are on the ProxyOptions page and all the information is densely located.